tests/Contrib/Unit/OTLPGrpcExporterTest.php

```diff
@@ -4,15 +4,104 @@
-use PHPUnit\Framework\TestCase;
 
-class OTLPGrpcExporterTest extends TestCase
+class OTLPGrpcExporterTest extends MockeryTestCase
 {
-    public function testExporter()
+    public function testExporterHappyPath()
+    {
+        /** @var MockInterface&TraceServiceClient */
+        $mockClient = Mockery::mock(TraceServiceClient::class)
+                        ->shouldReceive('Export')
+                        ->withArgs(function ($request) {
+                            return (count($request->getResourceSpans()) == 1);
+                        })
+                        ->andReturn(
+                            Mockery::mock(UnaryCall::class)
+                                ->shouldReceive('wait')
+                                ->andReturn(
+                                    [
+                                        'unused response data',
+                                        new class() {
+                                            public $code;
+                        
+                                            public function __construct()
+                                            {
+                                                $this->code = \Grpc\STATUS_OK;
+                                            }
+                                        },
+                                    ]
+                                )
+                                ->getMock()
+                        )
+                        ->getMock();
+        $exporter = new Exporter(
+            //These first parameters were copied from the constructor's default values
+            'localhost:4317',
+            true,
+            '',
+            '',
+            false,
+            10,
+            $mockClient
+        );
+               
+        $exporterStatusCode = $exporter->export([new SpanData()]);
+
+        $this->assertEquals(Exporter::SUCCESS, $exporterStatusCode);
+    }
+
+    public function testExporterUnexpectedGrpcResponseStatus()
+    {
+        /** @var MockInterface&TraceServiceClient */
+        $mockClient = Mockery::mock(TraceServiceClient::class)
+                        ->shouldReceive('Export')
+                        ->withArgs(function ($request) {
+                            return (count($request->getResourceSpans()) == 1);
+                        })
+                        ->andReturn(
+                            Mockery::mock(UnaryCall::class)
+                                ->shouldReceive('wait')
+                                ->andReturn(
+                                    [
+                                        'unused response data',
+                                        new class() {
+                                            public $code;
+                        
+                                            public function __construct()
+                                            {
+                                                $this->code = 'An unexpected status';
```

Comment:
Thoughts on moving some of this logic to private functions? E.g. one to get the mock client where you can pass in status, and another to get the exporter instance?

Reply:
Yeah I think I was leaning onto the [AHA (Avoid Hasty Abstraction)](https://kentcdodds.com/blog/aha-testing) side of things for this, but I can go ahead and split them off into a little helper

Reply:
I went ahead and split the mock creation into a helper and definitely think they read a lot nicer now, but I wasn't sure if splitting the Exporter constructor off into one would be over-doing it. Don't have a strong opinion either way so just lmk what y'all are feeling on this!

Reply:
Oh wow, that's pretty fancy :P I was just thinking like `private function createMockTraceServiceClient(int $statusCode)` and call it a day :sweat_smile:.